python/master.py

```diff
@@ -0,0 +1,321 @@
+            Rmat[i][j] = correlation[0]
+            Pmat[i][j]
+            j = j+1
+        i = i+1
+        
+    Rdf = pd.DataFrame(Rmat)
+    Rdf.columns = histone_list
+    Rdf.insert(0, 'Genes', gene_list, True)
+    Rdf = Rdf.set_index('Genes')
+    
+    Pdf = pd.DataFrame(Pmat)
+    Pdf.columns = histone_list
+    Pdf.insert(0, 'Genes', gene_list, True)
+    Pdf = Pdf.set_index('Genes')
+    return Rdf, Pdf
+
+
+def heatmap(data, ylabels, size, colour, title):
+    """Using dataframes based on the pearson_dfs fucntion, this function will create a simple seaborn heatmap. By using a dataframe the xlabels will be automatically created from the dataframes columns
+    """
+    plt.figure(figsize=size)
+    plt.xlabel('Histone Markers')
+    plt.title(title)
+    ax = sns.heatmap(data, cmap = colour, square = True, yticklabels = ylabels)
+    ax.set(xlabel = 'Histone Markers')
+    plt.show
+
+
+
+    
+def plotlyheat(df, size, title, xaxis, yaxis):
+    """ This function will take in a dataframe and create a plotly heatmap based on that dataframe. This is a good tool for gene lists that are very big and are hard to visulize on seaborn. 
+    
+    The dataframe used should be one made with the pearson_dfs. The size arugment is only necessary for really big gene lists. If the list is small the graph will be created using autosizing. 
+    
+    The resulting image will open up in google chrome. 
+    """
+    pio.renderers.default = "chrome"
+    fig = go.Figure(
+    data=(go.Heatmap(z = df, x =xaxis, y = yaxis, colorscale = "rdbu")),
+    layout_title_text=title)
+    
+    if size == None:
+        fig.update_layout(
+                autosize=True)
+    else:
+        fig.update_layout(
+                autosize=False,
+                width = size[0],
+                height = size[1])
+
+    fig.show()
+    
+def tissue_analysis(dictionary, common_cellline, histone_list, gene_list, data1, data2, name ):
+    """ This function is only necessary if you would like to perform some basic tissue analysis on data. The dictionary input should be a dictionary which matches cell lines to their specific tissue. 
+    
+    Common_cellline should be a list of the common_celllines between the two data sets. 
+    
+    Both data inputs should be dataframes created from the pearson_dfs function and the name will correspond to the type of tissue wishing to be analyzed. The output is a dataframe which can be graphed/
+    
+    This may not give you the best idea of tissue analysis as most datasets vary in the amount of celllines corresponding to each tissue. 
+    """
+    
+    df1 = data1.copy()
+    df2 = data2.copy()
+    for celllines in common_cellline:
+        if dictionary[celllines] != name:
+            del df1[celllines]
+            del df2[celllines]
+    matrix = pearson_dfs(gene_list, histone_list, df1, df2)[0]
+    return matrix
+
+"""
+CODE TO GET CCLE NORMALIZED
```

Comment:
If this code utilizes your functions, you should put this in a separate script (ie run_epigenetic_correlation.py) and call the functions in this module by using import master.

Reply:
How much would you like me to run in this file because I can run all the way through creating the heat maps to just simply creating 3 data frames (LeRoy, H3, and CCLE) or I can go anywhere in between I just dont know what is necessary